osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/TestAlarmSchedulerCommandExecutorTest.java

```diff
@@ -0,0 +1,198 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.openmuc.jdlms.AccessResultCode;
+import org.openmuc.jdlms.AttributeAddress;
+import org.openmuc.jdlms.ObisCode;
+import org.openmuc.jdlms.SetParameter;
+import org.openmuc.jdlms.datatypes.CosemDate;
+import org.openmuc.jdlms.datatypes.CosemTime;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.DlmsObjectConfigService;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.dlmsobjectconfig.model.DlmsObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.stub.DlmsConnectionManagerStub;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.stub.DlmsConnectionStub;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.dlms.interfaceclass.attribute.SingleActionScheduleAttribute;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.TestAlarmSchedulerRequestDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.TestAlarmTypeDto;
+import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
+
+@ExtendWith(MockitoExtension.class)
+class TestAlarmSchedulerCommandExecutorTest {
+
+  private static final Integer SINGLE_ACTION_CLASS_ID = 22;
+
+  private final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+  @InjectMocks private TestAlarmSchedulerCommandExecutor testAlarmSchedulerCommandExecutor;
+
+  @Mock private DlmsDevice device;
+
+  @Mock private TestAlarmSchedulerRequestDto testAlarmSchedulerRequestDto;
+
+  @Mock private MessageMetadata messageMetadata;
+
+  @Mock private DlmsObjectConfigService dlmsObjectConfigService;
+
+  @Mock private DlmsObject dlmsObject;
+
+  @Mock private ObisCode obisCode;
+
+  private DlmsConnectionManagerStub connectionManagerStub;
+  private DlmsConnectionStub connectionStub;
+
+  @BeforeEach
+  void setup() {
+
+    this.connectionStub = new DlmsConnectionStub();
+    this.connectionManagerStub = new DlmsConnectionManagerStub(this.connectionStub);
+
+    this.connectionStub.setDefaultReturnValue(DataObject.newArrayData(Collections.emptyList()));
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "EMPTY,LAST_GASP,No scheduled date-time set", //
+    "EMPTY,PARTIAL_POWER_OUTAGE,No scheduled date-time set",
+    "2020-10-07 15:33:00,PARTIAL_POWER_OUTAGE,Incorrect scheduled date time value set. It should not be a past date",
+    "2020-10-07 15:33:00,LAST_GASP,Incorrect scheduled date time value set. It should not be a past date"
```

Comment:
Remove `//` at end of first line.
Consider adding a test for an empty alarmtype as well.

Reply:
Added the suggested test cases and removed the // formatting.